pkg/compose/down.go

```diff
@@ -190,17 +201,108 @@ func (s *composeService) removeNetwork(ctx context.Context, name string, w progr
 	return nil
 }
 
-func (s *composeService) getServiceImagesToRemove(options api.DownOptions, project *types.Project) map[string]struct{} {
-	images := map[string]struct{}{}
+//nolint:gocyclo
```

Comment:
😬

Reply:
Limits, like rules, are meant to be broken!!

Just kidding - good call re: `ImageRemover` helper struct, I'd contemplated something like that, will do some refactoring now